File: .submodules/nostos/internal/cli/status.go
package cli

import (
	"fmt"
	"time"

	"github.com/spf13/cobra"

	"github.com/yurifrl/nostos/internal/registry"
)

func newStatusCmd() *cobra.Command {
	return &cobra.Command{
		Use:   "status",
		Short: "Show per-node reachability + Talos version",
		RunE: runEFuncSimple(func(cmd *cobra.Command, args []string) error {
			cfg, _, err := loadConfig()
			if err != nil {
				return err
			}
			entries := registry.List(cfg)
			rows := make([]registry.NodeStatus, 0, len(entries))
			for _, e := range entries {
				s := registry.Probe(e.Node, 1500*time.Millisecond)
				s.Name = e.Name
				rows = append(rows, s)
			}
			if outputMode == "json" {
				return outputJSON(rows)
			}
			fmt.Fprintf(cmd.OutOrStdout(), "Cluster %s\n", cfg.Cluster.Name)
			fmt.Fprintf(cmd.OutOrStdout(), "%-10s  %-16s  %-12s  %-8s  %-8s  %s\n",
				"NAME", "IP", "ROLE", "PING", "APID", "VERSION")
			for _, s := range rows {
				ver := s.Version
				if ver == "" {
					ver = "—"
				}
				fmt.Fprintf(cmd.OutOrStdout(), "%-10s  %-16s  %-12s  %s  %s  %s\n",
					s.Name, s.IP, s.Role, pill(s.Ping), pill(s.Apid), ver)
			}
			return nil
		}),
	}
}

File: .submodules/nostos/src/nostos/cluster/status.py
"""Per-node reachability + version + kubelet status report."""

from __future__ import annotations

from dataclasses import dataclass, field

from ..config import NostosConfig
from ..registry import NodeStatus, probe_node


@dataclass
class ClusterStatus:
    nodes: list[NodeStatus] = field(default_factory=list)


def cluster_status(cfg: NostosConfig, *, timeout: float = 2.0) -> ClusterStatus:
    status = ClusterStatus()
    for name, node in cfg.nodes.items():
        probe = probe_node(node, timeout=timeout)
        probe.name = name
        status.nodes.append(probe)
    return status
